packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -25,7 +25,7 @@ import { useEffect, useRef } from '@web-stories-wp/react';
 import { useHistory } from '../../history';
 
 // Record any change to core variables in history (history will know if it's a replay)
-function useHistoryEntry({ story, current, pages, selection }) {
+function useHistoryEntry({ story, current, pages, selection, capabilities }) {
```

Comment:
@swissspidy  This was why the taxonomy panel was disappearing after undoing something. 

So in the the storyProvider where `useHistoryEntry` effect is used, `capabilities` are being passed in but they weren't being accessed here so they'd just get lost to the story on undo. I suppose until now this hasn't been an issue. Should be all fixed though.

Reply:
Context https://github.com/google/web-stories-wp/issues/630 https://github.com/google/web-stories-wp/pull/1468

Reply:
CC @miina for extra context on this matter.

Reply:
@swissspidy @BrittanyIRL 

We had quite a long discussion on this about some point, resulting with conclusion that capabilities changing is not something that should be recorded as a history entry.

When the capabilities were included in the history, switching to the Document panel caused a new history entry (since capabilities were updated). This is actually not a real change in the user editing history though.

Removal issue: #630  

I haven't checked the code in the PR but just wanted to confirm there's no regression for capabilities creating a new history entry.

Thoughts?

Reply:
That's good to know. We're having to handle something similar when updating taxonomy to avoid redundant history updates. That is in PR #9147  ([here](https://github.com/google/web-stories-wp/pull/9147/files#diff-b0d9645393a6258bd68def9cad9560e65e01a48f7631bc87c6a1cb75ce344872R116-R123) and [here](https://github.com/google/web-stories-wp/pull/9147/files#diff-c4aa265ebf112ac8d5f1b42b1af4004298fd441f520c662e4176d32ce0b6c925R30-R36))


It's critical to taxonomies working that capabilities stay put in the stories object. It feels weird that they'd be part of the story object and then after any undo disappear.

Looking at https://github.com/google/web-stories-wp/issues/630 and how to replicate the issue it's still working as expected here while keeping capabilities synced with the story object. 

https://user-images.githubusercontent.com/10720454/134946802-f607a085-b827-4901-bc9e-c6abd9c14a80.mp4

Reply:
@miina let me know if there's any other worries here. I'll wait to merge it until I hear from you.

Reply:
@BrittanyIRL Thanks for the answer. If the history is working as expected then there are no worries :)